settings/DevHome.Settings/Views/AccountsPage.xaml.cs

```diff
@@ -58,7 +60,24 @@ private async void AddAccountButton_Click(object sender, RoutedEventArgs e)
         var numProviders = ViewModel.AccountsProviders.Count;
         if (numProviders == 1)
         {
-            await ShowLoginUIAsync("Settings", this, ViewModel.AccountsProviders[0]);
+            if (ViewModel.AccountsProviders[0].DeveloperIdProvider.GetAuthenticationExperienceKind() == Microsoft.Windows.DevHome.SDK.AuthenticationExperienceKind.CardSession)
+            {
+                await ShowLoginUIAsync("Settings", this, ViewModel.AccountsProviders[0]);
+            }
+            else if (ViewModel.AccountsProviders[0].DeveloperIdProvider.GetAuthenticationExperienceKind() == Microsoft.Windows.DevHome.SDK.AuthenticationExperienceKind.CustomProvider)
+            {
+                IntPtr windowHandle = IntPtr.Zero;
+                foreach (Process activeProcess in Process.GetProcesses())

```

Comment:
How would this react if another process had a window called "Dev Home"?

Reply:
I would think instead of instead of looking globally at all processes, shouldn't we be able to ask the system for our own process directly?

Reply:
If we just want the main Dev Home window handle, it's even simpler:
```cs
Application.Current.GetService<WindowEx>().GetWindowHandle();
```